src/appservice/MjolnirManager.ts

```diff
@@ -222,4 +246,56 @@ export class ManagedMjolnir {
     public get managementRoomId(): string {
         return this.mjolnir.managementRoomId;
     }
+
+    /**
+     * Intended to be called by the MjolnirManager to make sure the mjolnir is ready to listen to events.
+     * It should not be informed of any events via `onEvent` until this is called.
+     */
+    public async start(): Promise<void> {
+        await this.mjolnir.start();
+    }
+}
+
+export class MatrixIntentListener extends EventEmitter implements MatrixEmitter {
+    constructor(private readonly mjolnirId: string) {
+        super()
+    }
+
+    public handleEvent(mxEvent: WeakEvent) {
+        // These are ordered to be the same as matrix-bot-sdk's MatrixClient
+        // They shouldn't need to be, but they are just in case it matters.
+        if (mxEvent['type'] === 'm.room.member' &&  mxEvent.state_key === this.mjolnirId) {
+            if (mxEvent['content']['membership'] === 'leave') {
+                this.emit('room.leave', mxEvent.room_id, mxEvent);
+            }
+            if (mxEvent['content']['membership'] === 'invite') {
+                this.emit('room.invite', mxEvent.room_id, mxEvent);
+            }
+            if (mxEvent['content']['membership'] === 'join') {
+                this.emit('room.join', mxEvent.room_id, mxEvent);
+            }
+        }
+        if (mxEvent.type === 'm.room.message') {
+            this.emit('room.message', mxEvent.room_id, mxEvent);
+        }
+        if (mxEvent.type === 'm.room.tombstone' && mxEvent.state_key === '') {
+            this.emit('room.archived', mxEvent.room_id, mxEvent);
+        }
+        this.emit('room.event', mxEvent.room_id, mxEvent);
+
+    }
+
+    /**
+     * To be called by `Mjolnir`.
+     */
+    public async start() {
+
```

Comment:
Nit: If it's meant to do nothing, could you add a comment?
```
  // Nothing to do
```

Otherwise, it looks like you forgot to write the code.